src/v/cluster/tm_stm.h

```diff
@@ -79,6 +79,31 @@ struct tm_transaction {
     std::vector<tx_group> groups;
 
     friend std::ostream& operator<<(std::ostream&, const tm_transaction&);
+
+    std::string_view get_status() const {
+        switch (status) {
+        case tx_status::ongoing:
+            return "ongoing";
+        case tx_status::preparing:
+            return "preparing";
+        case tx_status::prepared:
+            return "prepared";
+        case tx_status::aborting:
+        case tx_status::killed:
+            return "aborting";
```

Comment:
The motivation behind mapping `killed` to `aborting` is to make user not to think about the subtle differences between both statuses. The problem is that `get_status` looks like a generic function which may be used in different contexts but because we map `killed` to `aborting` it makes it aware about the context we're going to use it in: to map status to something which is being send to user.

Some of the ideas how to fix this situation:
 - move the `get_status` out of `tm_stm` to `admin_server` (limit the context where is may be used)
 - keep it in `tm_stm` and translate `killed` to `killed` but update the tx object from `admin_server` to replace `killed` with `aborting` before calling `get_status`

Reply:
I added 
```
              // The motivation behind mapping killed to aborting is to make
              // user not to think about the subtle differences between both
              // statuses
              if (tx.status == cluster::tm_transaction::tx_status::killed) {
                  tx.status = cluster::tm_transaction::tx_status::aborting;
              }
```
to admin_api.cc